src/connection_string.ts

```diff
@@ -3,6 +3,7 @@ import * as fs from 'fs';
 import ConnectionString from 'mongodb-connection-string-url';
 import { URLSearchParams } from 'url';
 
+import type { OneOrMore } from '.';
```

Comment:
```suggestion
import type { OneOrMore } from './mongo_types';
```

I'm not sure if we could get a lint rule to help us check this? Maybe the forbidden imports thing? but we could defer that